app/src/main/java/eu/kanade/tachiyomi/util/AniSkipApi.kt

```diff
@@ -0,0 +1,149 @@
+        Log.i("bruh", "mal : $malId, num : $episodeNumber, len : $episodeLength")
+        val url =
+            "https://api.aniskip.com/v2/skip-times/$malId/$episodeNumber?types[]=ed&types[]=mixed-ed&types[]=mixed-op&types[]=op&types[]=recap&episodeLength=$episodeLength"
+        return try {
+            val a = client.newCall(GET(url)).execute().body!!.string()
+            Log.i("bruh", "res: $a")
+            val res = json.decodeFromString<AniSkipResponse>(a)
+            if (res.found) res.results else null
+        } catch (e: Exception) {
+            null
+        }
+    }
+
+    fun getMalIdFromAL(id: Long): Long {
+        val query = """
+                query{
+                Media(id:$id){idMal}
+                }
+        """.trimMargin()
+        val payload = buildJsonObject {
+            put("query", query)
+        }
+        val response = client.newCall(
+            POST(
+                "https://graphql.anilist.co",
+                body = payload.toString().toRequestBody(jsonMime),
+            ),
+        ).execute()
+        // i need to change this
+        val malId = response.body!!.string().substringAfter("idMal\":").substringBefore("}")
+        return malId.toLong()
+    }
+
+    class PlayerUtils(
+        private val binding: PlayerActivityBinding,
+        private val aniSkipResponse: List<Stamp>,
+    ) {
+        private val playerControls get() = binding.playerControls
+        private val activity: PlayerActivity get() = binding.root.context as PlayerActivity
+
+        fun showSkipButton(skipType: SkipType) {
+            val skipButtonString = when (skipType) {
+                SkipType.ed -> R.string.player_aniskip_ed
+                SkipType.op -> R.string.player_aniskip_op
+                SkipType.recap -> R.string.player_aniskip_recap
+                SkipType.mixedOp -> R.string.player_aniskip_mixedOp
+            }
+            launchUI {
+                playerControls.binding.controlsSkipIntroBtn.text = activity.getString(skipButtonString)
+            }
+        }
+
+        // this is used when netflixStyle is enabled
+        @SuppressLint("SetTextI18n")
+        fun showSkipButton(skipType: SkipType, waitingTime: Int) {
+            val skipTime = when (skipType) {
+                SkipType.ed -> aniSkipResponse.first { it.skipType == SkipType.ed }.interval
+                SkipType.op -> aniSkipResponse.first { it.skipType == SkipType.op }.interval
+                SkipType.recap -> aniSkipResponse.first { it.skipType == SkipType.recap }.interval
+                SkipType.mixedOp -> aniSkipResponse.first { it.skipType == SkipType.mixedOp }.interval
+            }
+            if (waitingTime > -1) {
+                if (waitingTime > 0) {
+                    launchUI {
+                        playerControls.binding.controlsSkipIntroBtn.text = activity.getString(R.string.player_aniskip_dontskip) + " " + waitingTime
```

Comment:
i think it looks a bit weird when there's just a normal number next to the text.
the user can't really understand that it's a countdown.
it probably needs a little more space and a different text style

Reply:
https://user-images.githubusercontent.com/80992641/191633075-3f0f3351-3148-4f10-9a68-02025cfcc7db.mp4

Reply:
I think that's better